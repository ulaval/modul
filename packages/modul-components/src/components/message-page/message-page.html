<article class="m-message-page"
         :class="{'m--is-state-confirmation': isStateConfirmation,
                  'm--is-state-information': isStateInformation,
                  'm--is-state-warning': isStateWarning,
                  'm--is-state-error': isStateError,
                  'm--is-skin-default': isSkinDefault,
                  'm--is-skin-light': isSkinLight }">
    <header class="m-message-page__header">
        <h2 class="m-message-page__title"
            v-if="title"
            v-html="title"></h2>
        <div class="m-message-page__icon-container"
             aria-hidden="true">
            <m-svg v-if="svgName"
                   class="m-message-page__icon"
                   :width="propImageSize"
                   :height="propImageSize"
                   :name="svgName" />
            <m-icon v-else
                    class="m-message-page__icon"
                    :name="iconNameProp"
                    :size="propImageSize" />
        </div>
    </header>
    <div class="m-message-page__body"
         v-if="hasBody">
        <p class="m-message-page__hints"
           :class="{'m--no-content': !hasLinksAndSlot}"
           v-if="hasHints"
           v-for="hint in hints"
           ref="hint">{{hint}}</p>
        <div class="m-message-page__details"
             v-if="$slots.default">
            <slot />
        </div>
        <ul class="m-message-page__links"
            v-if="hasLinks">
            <li ref="link"
                v-for="link in links">
                <m-link class="m-message-page__link"
                        bullet-point="true"
                        mode="link"
                        :url="link.url"
                        :target="isTargetExternal(link.external)">{{link.label}}</m-link>
            </li>
        </ul>
    </div>
</article>
