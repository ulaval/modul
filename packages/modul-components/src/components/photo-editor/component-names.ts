export const CROP_IMAGE_NAME: string = 'm-crop-image';
export const IMAGE_DISPLAYER_NAME: string = 'm-image-displayer';
