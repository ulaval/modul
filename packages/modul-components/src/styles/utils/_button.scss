.mu-button-primary {
    @include m-button();
    @include m-button--primary();
    @include m-button--hover();
}

.mu-button-secondary {
    @include m-button();
    @include m-button--secondary();
    @include m-button--hover();
}

.mu-button-disabled {
    @include m-button();
    @include m-button--disabled();
}

.mu-button-waiting {
    @include m-button();
    @include m-button--waiting();
}
