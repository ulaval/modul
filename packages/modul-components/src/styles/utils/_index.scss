@import 'utils.mixin';

@include mu-utils();
