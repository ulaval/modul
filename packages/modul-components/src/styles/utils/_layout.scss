$m-position-map: (
    '': '',
    't': '-top',
    'r': '-right',
    'b': '-bottom',
    'l': '-left',
);

$m-space-map: (
    '-4xl': $m-space-4xl,
    '-3xl': $m-space-3xl,
    '-2xl': $m-space-2xl,
    '-xl': $m-space-xl,
    '-lg': $m-space-lg,
    '-md': $m-space-md,
    '' : $m-space,
    '-sm': $m-space-sm,
    '-xs': $m-space-xs,
    '-2xs': $m-space-2xs
);

@mixin m-space-factory($attribute, $has-no-value: false) {
    @each $key-position, $value-position in $m-position-map {
        @if $has-no-value == true {
            &#{$key-position} {
                #{$attribute}#{$value-position}: 0;
            }
        } @else {
            @each $key-space, $value-space in $m-space-map {
                &#{$key-position}#{$key-space} {
                    #{$attribute}#{$value-position}: #{$value-space};
                }
            }
        }

    }
}

.mu {
    // Margin
    &-m {
        @include m-space-factory(margin);
    }

    &-no-m {
        @include m-space-factory(margin, true);
    }

    // Padding
    &-p {
        @include m-space-factory(padding);
    }

    &-no-p {
        @include m-space-factory(padding, true);
    }

    // Display
    &-d {
        &-block {
            display: block;
        }

        &-flex {
            display: flex;
        }

        &-inline {
            display: inline;

            &-block {
                display: inline-block;
            }

            &-flex {
                display: inline-flex;
            }
        }
    }

    // Max and min width
    &-max-w {
        max-width: $m-max-width;

        &-xs {
            max-width: $m-max-width-xs;
        }

        &-sm {
            max-width: $m-max-width-sm;
        }

        &-lg {
            max-width: $m-max-width-lg;
        }

        &-text {
            max-width: $m-max-width-text;
            display: flex !important;
        }
    }

    &-min-w {
        min-width: $m-min-width;
    }
}
