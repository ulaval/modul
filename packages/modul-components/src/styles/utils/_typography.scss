.mu-font {
    &-size {
        @include m-font-size();

        &-lg {
            @include m-font-size('lg');
        }

        &-md {
            @include m-font-size('md');
        }

        &-sm {
            @include m-font-size('sm');
        }

        &-xs {
            @include m-font-size('xs');
        }
    }

    &-style {
        &-normal {
            font-style: normal;
        }

        &-italic {
            font-style: italic;
        }
    }

    &-weight {
        &-black {
            font-weight: $m-font-weight-black;
        }

        &-bold {
            font-weight: $m-font-weight-bold;
        }

        &-semi-bold {
            font-weight: $m-font-weight-semi-bold;
        }

        &-regular {
            font-weight: $m-font-weight-regular;
        }

        &-light {
            font-weight: $m-font-weight-light;
        }
    }
}

.mu-typo {
    &-precision {

        &,
        &-italic {
            @include m-font-size('sm');
            color: $m-color-text-light;
        }

        &-italic {
            font-style: italic;
        }
    }

    &-highlighting {
        display: inline;
        @include m-font-size('sm');
        font-weight: $m-font-weight-semi-bold;
        background: $m-color-warning-lightest;
        box-shadow: 0 0 0 2px $m-color-warning-lightest;
    }
}
