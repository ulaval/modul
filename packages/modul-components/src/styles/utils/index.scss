@import 'typographies';
@import 'content';
@import 'layout';
@import 'visually-hidden';
