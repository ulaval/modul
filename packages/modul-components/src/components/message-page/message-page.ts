import { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';
import { ModulIconName } from '../../utils/modul-icons/modul-icons';
import { ModulVue } from '../../utils/vue/vue';
import { MESSAGE_PAGE_NAME } from '../component-names';
import IconPlugin from '../icon/icon';
import LinkPlugin from '../link/link';
import { MMessageState } from '../message/message';
import ModalPlugin from '../modal/modal';
import { MSvg } from './../svg/svg';
import WithRender from './message-page.html?style=./message-page.scss';

/**
 * Utility class to manage the properties related to the link displayed in the error pages.
 */
export class Link {
    /**
     * Constructor
     * @param label Label for the url link to display.
     * @param url Target for the location to navigate to, can be relative.
     * @param external Defines to the target is external (opens in new tab), or internal (opens in same tab) to the application.
     */
    constructor(public label: string, public url: string, public external: boolean = false) {
    }
}

export enum MMessagePageSkin {
    Default = 'default',
    Light = 'light'
}

export enum MMessagePageImageSize {
    Default = '130px',
    Small = '76px'
}

@WithRender
@Component({
    components: { MSvg }
})
export class MMessagePage extends ModulVue {

    @Prop({
        default: 'error',
        validator: value =>
            value === MMessageState.Information ||
            value === MMessageState.Warning ||
            value === MMessageState.Confirmation ||
            value === MMessageState.Error
    })
    public state: string;

    @Prop({
        default: MMessagePageSkin.Default,
        validator: value =>
            value === MMessagePageSkin.Default ||
            value === MMessagePageSkin.Light
    })
    public skin: string;

    @Prop()
    public iconName: string;

    @Prop({})
    public svgName: string;

    @Prop()
    public imageSize: string;

    @Prop()
    public title: string;

    @Prop({ default: () => [] })
    public hints: string[];

    @Prop({ default: () => [] })
    public links: Link[];

    public get hasHints(): boolean {
        return this.hints.length > 0;
    }

    public get hasLinks(): boolean {
        return this.links.length > 0;
    }

    public get propImageSize(): string {
        if (this.imageSize) {
            return this.imageSize;
        }
        return this.isSkinLight ? MMessagePageImageSize.Small : MMessagePageImageSize.Default;
    }

    public isTargetExternal(isExternal: boolean): string {
        return isExternal ? '_blank' : '';
    }

    public get hasLinksAndSlot(): boolean {
        return this.hasLinks || !!this.$slots.default;
    }

    public get hasBody(): boolean {
        return this.hasHints || this.hasLinks || !!this.$slots.default;
    }

    public get isSkinDefault(): boolean {
        return this.skin === MMessagePageSkin.Default;
    }

    public get isSkinLight(): boolean {
        return this.skin === MMessagePageSkin.Light;
    }

    public get isStateInformation(): boolean {
        return this.state === MMessageState.Information;
    }

    public get isStateWarning(): boolean {
        return this.state === MMessageState.Warning;
    }

    public get isStateError(): boolean {
        return this.state === MMessageState.Error;
    }

    public get isStateConfirmation(): boolean {
        return this.state === MMessageState.Confirmation;
    }

    public get iconNameProp(): string {
        if (this.iconName) {
            return this.iconName;
        } else {
            switch (this.state) {
                case MMessageState.Confirmation:
                    return ModulIconName.ConfirmationWhiteFilled;
                case MMessageState.Information:
                    return ModulIconName.InformationWhiteFilled;
                case MMessageState.Warning:
                    return ModulIconName.WarningWhiteFilled;
                default:
                    return ModulIconName.ErrorWhiteFilled;
            }
        }
    }
}

const MessagePagePlugin: PluginObject<any> = {
    install(v): void {
        v.prototype.$log.debug(MESSAGE_PAGE_NAME, 'plugin.install');
        v.use(LinkPlugin);
        v.use(IconPlugin);
        v.use(ModalPlugin);
        v.component(MESSAGE_PAGE_NAME, MMessagePage);
    }
};

export default MessagePagePlugin;
