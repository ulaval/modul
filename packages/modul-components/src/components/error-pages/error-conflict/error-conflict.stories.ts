import { array, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/vue';
import Vue from 'vue';
import { componentsHierarchyRootSeparator } from '../../../../conf/storybook/utils';
import { ERROR_CONFLICT_NAME } from '../../component-names';
import { Link } from '../../message-page/message-page';
import ErrorConflictPlugin from './error-conflict';

Vue.use(ErrorConflictPlugin);

storiesOf(`${componentsHierarchyRootSeparator}/error-pages/${ERROR_CONFLICT_NAME}`, module)


    .add('default', () => ({
        template: `<div style="border: solid 1px black; padding: 10px; width: 600px;">
                        <m-error-conflict></m-error-conflict>
                   </div>`
    }))
    .add('title', () => ({
        props: {
            title: {
                default: text('Title', 'A Custom Title')
            }
        },
        template: `<div style="border: solid 1px black; padding: 10px; width: 600px;">
                        <m-error-conflict :title="title"></m-error-conflict>
                   </div>`
    }))
    .add('links', () => ({
        props: {
            links: {
                default: array('Links[Array]', [new Link('The first custom link', 'http://www.ulaval.ca', true), new Link(`The second custom
                link`, 'http://www.google.com', true)])
            }
        },
        template: `<div style="border: solid 1px black; padding: 10px; width: 600px;">
                        <m-error-conflict :links="links"></m-error-conflict>
                   </div>`
    }))
    .add('hints', () => ({
        props: {
            hints: {
                default: array('Hints[Array]', ['My only custom hint', `My second (long) custom hint.Lorem ipsum dolor
                sit amet, consectetur adipiscing elit. Nulla egestas urna rhoncus ipsum congue lobortis. `])
            }
        },
        template: `<div style="border: solid 1px black; padding: 10px; width: 600px;">
                        <m-error-conflict :hints="hints"></m-error-conflict>
                   </div>`
    }));
