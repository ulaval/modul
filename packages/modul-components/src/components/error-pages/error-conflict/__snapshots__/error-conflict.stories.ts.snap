// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots components|/error-pages/m-error-conflict default 1`] = `
<div
  style="border: 1px solid black; padding: 10px; width: 600px;"
>
  <article
    class="m-message-page m--is-state-warning m--is-skin-default"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        La donnée sur laquelle vous travaillez a été modifiée par un autre utilisateur
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <div
          class="m-message-page__icon"
          style="width: 130px;"
        >
          {}
        </div>
      </div>
    </header>
     
    <div
      class="m-message-page__body"
    >
      <p
        class="m-message-page__hints m--no-content"
      >
        Vos modifications n'ont pas été enregistrées et les données ont été rafraichies. Veuillez reprendre votre travail.
      </p>
       
      <!---->
       
      <!---->
    </div>
  </article>
</div>
`;

exports[`Storyshots components|/error-pages/m-error-conflict hints 1`] = `
<div
  style="border: 1px solid black; padding: 10px; width: 600px;"
>
  <article
    class="m-message-page m--is-state-warning m--is-skin-default"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        La donnée sur laquelle vous travaillez a été modifiée par un autre utilisateur
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <div
          class="m-message-page__icon"
          style="width: 130px;"
        >
          {}
        </div>
      </div>
    </header>
     
    <div
      class="m-message-page__body"
    >
      <p
        class="m-message-page__hints m--no-content"
      >
        My only custom hint
      </p>
      <p
        class="m-message-page__hints m--no-content"
      >
        My second (long) custom hint.Lorem ipsum dolor
                sit amet, consectetur adipiscing elit. Nulla egestas urna rhoncus ipsum congue lobortis. 
      </p>
       
      <!---->
       
      <!---->
    </div>
  </article>
</div>
`;

exports[`Storyshots components|/error-pages/m-error-conflict links 1`] = `
<div
  style="border: 1px solid black; padding: 10px; width: 600px;"
>
  <article
    class="m-message-page m--is-state-warning m--is-skin-default"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        La donnée sur laquelle vous travaillez a été modifiée par un autre utilisateur
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <div
          class="m-message-page__icon"
          style="width: 130px;"
        >
          {}
        </div>
      </div>
    </header>
     
    <div
      class="m-message-page__body"
    >
      <p
        class="m-message-page__hints"
      >
        Vos modifications n'ont pas été enregistrées et les données ont été rafraichies. Veuillez reprendre votre travail.
      </p>
       
      <!---->
       
      <ul
        class="m-message-page__links"
      >
        <li>
          <a
            class="m-link m-message-page__link m--has-left-icon m--has-content"
            href="http://www.ulaval.ca"
            tabindex="0"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-link__icon m--has-large-stroke"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <span
              class="m-link__text"
            >
              The first custom link
            </span>
             
            <span
              class="m-i18n m-link__hidden"
            >
              Cet hyperlien s'ouvrira dans un nouvel onglet.
            </span>
          </a>
        </li>
        <li>
          <a
            class="m-link m-message-page__link m--has-left-icon m--has-content"
            href="http://www.google.com"
            tabindex="0"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="m-icon m-link__icon m--has-large-stroke"
              height="12px"
              width="12px"
            >
              <!---->
               
              <use
                aria-hidden="true"
                class=""
              />
            </svg>
             
            <span
              class="m-link__text"
            >
              The second custom
                link
            </span>
             
            <span
              class="m-i18n m-link__hidden"
            >
              Cet hyperlien s'ouvrira dans un nouvel onglet.
            </span>
          </a>
        </li>
      </ul>
    </div>
  </article>
</div>
`;

exports[`Storyshots components|/error-pages/m-error-conflict title 1`] = `
<div
  style="border: 1px solid black; padding: 10px; width: 600px;"
>
  <article
    class="m-message-page m--is-state-warning m--is-skin-default"
  >
    <header
      class="m-message-page__header"
    >
      <h2
        class="m-message-page__title"
      >
        A Custom Title
      </h2>
       
      <div
        aria-hidden="true"
        class="m-message-page__icon-container"
      >
        <div
          class="m-message-page__icon"
          style="width: 130px;"
        >
          {}
        </div>
      </div>
    </header>
     
    <div
      class="m-message-page__body"
    >
      <p
        class="m-message-page__hints m--no-content"
      >
        Vos modifications n'ont pas été enregistrées et les données ont été rafraichies. Veuillez reprendre votre travail.
      </p>
       
      <!---->
       
      <!---->
    </div>
  </article>
</div>
`;
